src/cpr_data_access/models.py

```diff
@@ -1097,3 +1105,105 @@ def to_huggingface(
         )
 
         return huggingface_dataset
+
+    def from_huggingface(
+        self,
+        huggingface_dataset: HFDataset,
+        limit: Optional[int] = None,
+    ) -> "Dataset":
+        """
+        Create a dataset from a huggingface dataset.
+
+        :param huggingface_dataset: created using `Dataset.to_huggingface()`
+        :param limit: optionally limit the number of documents to load
+        :return self: with documents loaded from huggingface dataset
+        """
+
+        document_metadata_model = self.document_model.__fields__[
+            "document_metadata"
+        ].outer_type_
+        hf_dataframe: pd.DataFrame = huggingface_dataset.to_pandas()
+
+        if limit is not None:
+            doc_ids = hf_dataframe["document_id"].unique()[:limit]
+            hf_dataframe = hf_dataframe[hf_dataframe["document_id"].isin(doc_ids)]
+
+        documents = []
+
+        for _, doc_df in tqdm(
+            hf_dataframe.groupby("document_id"),
+            total=hf_dataframe["document_id"].nunique(),
+            unit="docs",
+        ):
+            document_text_blocks = [
+                TextBlock(
+                    # TODO: we aren't able to access the original text split over lines as it's not stored in the huggingface dataset
+                    text=[row["text"]],
+                    text_block_id=row["text_block_id"],
+                    language=row["language"],
+                    type=row["type"],
+                    type_confidence=row["type_confidence"],
+                    page_number=row["page_number"],
+                    coords=[tuple(c) for c in row["coords"].tolist()] if row["coords"] is not None else None,  # type: ignore
+                )
+                for _, row in doc_df.iterrows()
+            ]
+
+            # pandas to_dict() stores sequences as numpy arrays, so we need to convert them back to lists
+            doc_fields = {
+                k: list(v) if isinstance(v, np.ndarray) else v
+                for k, v in doc_df.iloc[0].to_dict().items()
```

Comment:
Why are we only taking the `.iloc[0]` here?

Reply:
Is this based on the assumption above that there's 1 document per dataframe? Is that true for translated documents that have the same `document_id` as the original?

If we're assuming this, I'd put an assertion above to make it explicit -- if for some reason we'll get multiple, this will be hard to identify.

Reply:
This is based on the assumption that all documents share the same data model - so the columns of the first row of the dataframe are representative of all the documents in the dataset. If you're happy with this logic I can make it clearer with a comment?

Reply:
I see -- I would still be cautious of disregarding all other rows except the first. This makes me think that we could store this as metadata of the whole dataset then, and that would make the assumption clear? 

Or if we keep it this way, to add an assertion about them being the same indeed -- if in the future we'd want a full corpus with GST and CPR data all in one place this wouldn't allow to make that mistake.

Reply:
Ah, sorry – I was forgetting about the groupby document step. This makes the assumption that the document metadata for each text block within a document is the same.

For us to check that all the values of just the _document metadata_ fields are the same (rather than also the metadata fields related to text blocks, which will be different) this means accessing the `__fields__` property of the type of `DocumentMetadata` e.g. `CPRDocumentMetadata.__fields__`. The only other alternative I can see is processing each row through the `DocumentMetadata` model which would slow things down significantly. 

If you're happy with this or can see another way i'm happy to implement as this seems sensible.